Create rake task for sending PIN confirmation emails

BryanEliDimas opened the pull request:
Used for testing the deliverability of PIN confirmation emails.

Creates the rake command `rake send_pin_confirmation_emails`

It can be run this way:

`EMAILS_CSV_PATH='emails_list.csv' rake send_pin_confirmation_emails`

Needs a CSV file with only one column and no headers, for example: 

[emails_list.csv](#)
```
bryan@me.com
eli@me.com
bryaneli@me.com
```

Dantemss commented on lib/tasks/test_pin_email_deliverability.rake:
```diff
@@ -0,0 +1,17 @@
+        confirmation_pin: TokenMaker.contact_info_confirmation_pin,
+        confirmation_token: TokenMaker.contact_info_confirmation_code,
+        user_full_name: Faker::Name.name
+    )
```
Something (either this task or the method) needs to call either `deliver_now` or `deliver_later` on the output of `mail`.

BryanEliDimas replied:
DOH. How did I miss that. Fixed.

BryanEliDimas commented:
> re: Rather than making a new mailer and template, couldn't we just make the task reuse the existing ConfirmationMailer?

No, because that requires a `ContactInfo` for each email and each one requires a `User` and stuff. 

Addressed your comment, force-pushed the changes. Please review again with the changes?

Dantemss commented:
In that case I would suggest that inside a transaction we create the new Users/ContactInfos and use the real mailer, then ensure we rollback the transaction at the end. The goal is to minimize the amount of test-specific code so the test can be closer to the real thing.

But maybe @jpslav has other opinions?

BryanEliDimas commented:
Supa smart @Dantemss. With that suggestion, it turns out that we _will_ be able to run it directly through SSH instead of the release process! 🚀 

[EMAILS_CSV_PATH='emails_list.csv' bin/rails console](#)
irb >
```
CSV.foreach(ENV['EMAILS_CSV_PATH']) { |row|
  PreAuthState.transaction {
    pas = PreAuthState.create(is_partial_info_allowed: true, contact_info_value: row[0].try(:strip), role: 'student');
    puts SignupConfirmationMailer.instructions(pre_auth_state: pas).deliver_now;
    raise ActiveRecord::Rollback
  }
}
```

Dantemss commented:
Yes but test on staging etc first 😬This is a little scary to run directly on prod.

BryanEliDimas commented:
Closing because we ended up running the code I typed above directly via the rails console and it worked.

BryanEliDimas commented:
@scb6 New code because the analysts need all email to have the same subject line:

```
CSV.foreach(ENV['EMAILS_CSV_PATH']) { |row|
  PreAuthState.transaction {
    pas = PreAuthState.create(is_partial_info_allowed: true, contact_info_value: row[0].try(:strip), role: 'student');
    pas.update_attributes(confirmation_pin: 122608);
    puts(SignupConfirmationMailer.instructions(pre_auth_state: pas).deliver_now);
    raise ActiveRecord::Rollback
  }
}
```

Note that the PIN code for all email will be: 122608
